src/actions/mod.rs

```diff
@@ -812,6 +850,76 @@ impl ActionHandler {
     }
 }
 
+fn infer_config_defaults(project_dir: &Path, config: &mut Config) -> CargoResult<()> {
```

Comment:
Yeah, I think I would expect this to be in the cargo module and probably a method on `Config`. Taking a mut ref to Config is a bit strange, I guess I would expect to take a Config and return a new one? And that might make some of the referencing calling this easier? Anyway, I'm going to merge this now, we can refactor later if you like because I think this is polish.